File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_idx = nums[0];
        int cur_idx = 0;
        while(max_idx < nums.size() - 1  && cur_idx <= max_idx ){
            max_idx = max(cur_idx + nums[cur_idx] , max_idx);
            cur_idx++;
        }
        return (max_idx >= nums.size() - 1);
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.py
from typing import List

class Solution:
    def aux_LIS(self, nums: List[int], index: int) -> int:
        if index == len(nums) - 1:
            return 1
        
        max_list = [self.aux_LIS(nums, i) + 1 for i in range(index + 1, len(nums)) if nums[index] < nums[i]]
        max_list.append(1)  # If no increasing sequence is found, return 1.
        
        return max(max_list)
    
    def lengthOfLIS(self, nums: List[int]) -> int:
        return max(self.aux_LIS(nums, i) for i in range(len(nums)))  # Start LIS from each index



        
        